e2e/specs/age-based-vitals.spec.ts

```diff
@@ -0,0 +1,212 @@
+      });
+
+      await test.step(`And I enter the respiration rate: ${group.normalVitals.respiration}`, async () => {
+        await vitalsPage.page
+          .getByRole('spinbutton', { name: /respiration rate/i })
+          .fill(group.normalVitals.respiration);
+      });
+
+      await test.step('And I click on the `Save and close` button to save the vitals', async () => {
+        await vitalsPage.page.getByRole('button', { name: /save and close/i }).click();
+      });
+
+      await test.step('Then I should see a success notification confirming vitals were saved', async () => {
+        await expect(vitalsPage.page.getByText(/vitals and biometrics saved/i)).toBeVisible();
+      });
+
+      await test.step('And I should see the vitals table headers', async () => {
+        await expect(headerRow).toContainText(/temp/i);
+        await expect(headerRow).toContainText(/bp/i);
+        await expect(headerRow).toContainText(/r. rate/i);
+      });
+
+      await test.step(`And I should see the temperature value of ${group.normalVitals.temp} in the vitals table`, async () => {
+        await expect(dataRow).toContainText(group.normalVitals.temp);
+      });
+
+      await test.step(`And I should see the blood pressure value of ${group.normalVitals.systolic}/${group.normalVitals.diastolic} in the vitals table`, async () => {
+        await expect(dataRow).toContainText(`${group.normalVitals.systolic} / ${group.normalVitals.diastolic}`);
+      });
+
+      await test.step(`And I should see the respiration rate of ${group.normalVitals.respiration} in the vitals table`, async () => {
+        await expect(dataRow).toContainText(group.normalVitals.respiration);
+      });
+
+      await test.step('And the temperature cell should have normal styling', async () => {
+        const normalCell = vitalsPage.page.getByRole('cell', { name: group.normalVitals.temp });
+        const backgroundColor = await normalCell.evaluate((el) => window.getComputedStyle(el).backgroundColor);
+        expect(backgroundColor).toBe('rgb(255, 255, 255)');
+
+        const afterContent = await normalCell.evaluate((el) => {
+          const after = window.getComputedStyle(el, '::after');
+          return after.content;
+        });
+        expect(afterContent).toBe('none');
+      });
+    });
+
+    test(`Critical range vitals validation flagging for ${group.name} patient`, async ({ api, page }) => {
+      const birthdate = calculateBirthdate(group.age);
+      const patient = await generateRandomPatient(api, { birthdate });
+      createdPatients.push(patient);
+      const visit = await startVisit(api, patient.uuid);
+      createdVisits.push(visit);
+
+      const vitalsPage = new BiometricsAndVitalsPage(page);
+      const headerRow = vitalsPage.vitalsTable().locator('thead > tr');
+      const dataRow = vitalsPage.vitalsTable().locator('tbody > tr');
```

Comment:
This kind of stuff could be encoded in the `vitalsPage` as well, e.g., `vitalsPage.vitalsHeader()`, `vitalsPage.vitalsFirstRow()`